frontend/src/component/common/util.test.ts

```diff
@@ -114,4 +114,66 @@ describe('updateWeightEdit', () => {
         );
         expect(weights).toEqual([500, 500, 0]);
     });
+
+    describe('sum over 100% does not result in negative weight', () => {
+        it('can deal with weight of 2 items', () => {
+            const variants = [
+                {
+                    ...variantTemplate,
+                    weightType: 'fix' as const,
+                    weight: 600,
+                    id: '1',
+                    name: 'A',
+                },
+                {
+                    ...variantTemplate,
+                    weightType: 'fix' as const,
+                    weight: 600,
+                    id: '2',
+                    name: 'B',
+                },
+                { ...variantTemplate, id: '3', name: 'C' },
+            ];
+            const weights = updateWeightEdit(variants, 1000).map(
```

Comment:
can we add spaces between given/when/then sections so that we know what is test preparation/action execution/expectation checking?